Add Zoom OAuth Integration + Zoom Transcript Loader

bradtaylorsf opened the pull request:
# Zoom OAuth2 Integration PR

This PR introduces full **OAuth2 support for Zoom**, enabling users to authenticate their Zoom accounts and pull transcript recordings from selected Zoom meetings. It includes the UI, backend, and document loader components to support this integration end-to-end.

---

## ✅ Features Included

### 1. Zoom OAuth Credential Integration

* **New credential type:** `ZoomOAuth`
* **OAuth login flow** via `/zoom-auth` and `/zoom-auth/callback` routes
* **Authenticated user details + tokens** are stored in a credential object:

  * `zoomAccessToken`
  * `zoomRefreshToken`
  * `provider`, `providerId`, `email`, `fullName`

### 2. Zoom Transcript Document Loader

* **New loader:** Zoom Transcripts
* Allows users to **select Zoom meetings from the past 30 days**
* Downloads and parses `.vtt` transcript files from the meeting recordings
* Supports optional text splitting and metadata filtering
* Built-in handling for **access token refresh on expiry**

### 3. UI Components

* `ZoomAuthButton`: Trigger Zoom OAuth popup
* `ZoomMeetingPicker`: Dialog for selecting past meetings with available recordings
* Dynamic field handling inside the DocStore and Credential UI

---

## 📦 New Dependencies

*Added to `packages/server/package.json`:*

```json
"@giorgosavgeris/passport-zoom-oauth2": "^1.2.0"
```

---

## 🔐 New Environment Variables

Add the following to your `.env` file or hosting provider config:

```env
ZOOM_CLIENT_ID=your-zoom-client-id
ZOOM_CLIENT_SECRET=your-zoom-client-secret
ZOOM_CALLBACK_URL=https://yourdomain.com/api/v1/zoom-auth/callback
```

This PR includes a sample Zoom App configured at:
👉 [[Zoom Marketplace – Registered App](https://marketplace.zoom.us/develop/applications/DtYZJ0veRjCMo4BCuWDNbg/information)](https://marketplace.zoom.us/develop/applications/DtYZJ0veRjCMo4BCuWDNbg/information)

---

## 🔍 Local Testing with Ngrok

To test OAuth locally:

1. **Run your local server:**

   ```bash
   pnpm dev
   ```

2. **Start ngrok on the same port:**

   ```bash
   npx ngrok http 3000
   ```

3. **Copy the HTTPS ngrok URL and update your .env:**

   ```env
   ZOOM_CALLBACK_URL=https://your-ngrok-id.ngrok.io/api/v1/zoom-auth/callback
   ```

4. **Update your Zoom App’s Redirect URL for OAuth** in the [[Zoom App Marketplace](https://marketplace.zoom.us/develop/applications/DtYZJ0veRjCMo4BCuWDNbg/information)](https://marketplace.zoom.us/develop/applications/DtYZJ0veRjCMo4BCuWDNbg/information) to match your ngrok callback URL.

---

## 🧪 How to Test

1. Go to “Add Credential” → Choose “Zoom OAuth” → Click “Authorize with Zoom”.
2. Complete Zoom sign-in and approve access.
3. Use the “Zoom Transcripts” loader in the DocStore → Select meetings with transcripts.
4. Preview or Save to verify document ingestion works correctly.

---

bradtaylorsf committed:
Add Zoom transcript loader and OAuth credential

bradtaylorsf committed:
Zoom login and document loader

ct3685 commented:
BWS on AAI Local is showing the following vars, is BWS missing one? According to the PR here its stating that a callback url is required, is this different on staging and production and hard coded in so we're just using ngrok locally to check?

ZOOM_CLIENT_ID
ZOOM_CLIENT_SECRET
ZOOM_SECRET

bradtaylorsf commented:
It will need to be rebased, btw

bradtaylorsf committed:
chore: update pnpm-lock.yaml after rebase

bradtaylorsf committed:
chore: update pnpm-lock.yaml with latest staging dependencies

bradtaylorsf committed:
added zoom service updates and Auth0 bug

diecoscai committed:
feat: chief sidekick is auto-selected and loaded on login, chatflow list hidden unless no chief sidekick, and left navbar opens by default (closes #316)

bradtaylorsf committed:
added table style to the zoom meeting picker

diecoscai committed:
refactor: replace cookie-based defaultChatflowId with client-side API approach

This PR refactors the default chatflow redirection mechanism by removing
server-side cookie dependency and implementing a cleaner client-side solution.

## Changes Made

### Removed Cookie Implementation
- Removed defaultChatflowId cookie reading from getCachedSession.ts
- Removed cookie setting functionality from authentication middleware
- Eliminated server-side cookie dependency for default chatflow handling

### Implemented Client-Side Solution
- Added ChatRedirectHandler component for managing chat redirects
- Created useRedirectToDefaultChatflow hook for redirect logic
- Added auth API client for /auth/me endpoint calls
- Enhanced authentication middleware with direct /auth/me endpoint

### Architecture Improvements
- Replaced server-side cookie management with API-based approach
- Improved debugging capabilities with client-side logic
- Reduced security concerns related to cookie handling
- Simplified state management by removing cookie dependencies

## Benefits
- Better separation of concerns (client handles UI, server provides data)
- Easier debugging and testing of redirect logic
- No cookie security implications
- More maintainable and modern approach
- Cleaner code architecture

bradtaylorsf committed:
Fixed ts errors and added gtag

bradtaylorsf committed:
added alpha button for voice agents

diecoscai committed:
fix: resolve template save authorization error

- Add user context parameter to saveCustomTemplate service method
- Pass req.user from controller to service for proper authorization
- Add enforceAbility middleware to custom template route
- Fix 'Unauthorized public access to non-public chatflow' error

Resolves: Diego Ticket 1 - Template save functionality

diecoscai committed:
fix: improve template export and import functionality

- Simplify generateExportFlowData to pass complete chatflow object
- Add 404 error handling for missing chatflows
- Clean up marketplace-specific descriptions on import
- Improve error handling in canvas flow loading

Resolves: Diego Ticket 1 - Template export functionality
